### PR TITLE
dart: 2.12.2 -> 2.13.1

### PR DESCRIPTION
**Pull request by NANASHI0X74**
also use regular dart derivation in flutter again instead of an overridden one

###### Motivation for this change
updating stuff :p
###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

tested dart create and dart run, flutter derivation stayed the same (nixpkgs-review didn't recognize it as an updated package)

### CODE REVIEW

**Review by thiagokokada — APPROVED** [review]
Code-wise LGTM.

**Comment by thiagokokada** [comment]
Result of `nixpkgs-review pr 126979` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>dart</li>
  </ul>
</details>

**Review by thiagokokada — APPROVED** [review]

File: pkgs/development/compilers/flutter/default.nix [hunk 1]
```diff
@@ -5,39 +5,17 @@ let
...
   mkFlutter = mkFlutter;
   stable = mkFlutter rec {
-    dart = dartStable;
-    inherit version;
+    inherit dart version;
     pname = "flutter";
     src = fetchurl {
-      url = "https://storage.googleapis.com/flutter_infra/releases/${channel}/linux/${filename}";
+      url =
+        "https://storage.googleapis.com/flutter_infra/releases/${channel}/linux/${filename}";
```
Comment (SuperSandro2000) [inline pkgs/development/compilers/flutter/default.nix:18]:
```suggestion
      url = "https://storage.googleapis.com/flutter_infra/releases/${channel}/linux/${filename}";
```

File: pkgs/development/interpreters/dart/default.nix [hunk 2]
```diff
@@ -2,33 +2,36 @@
...
     # also need to override sources, to avoid mistakes
-    version = "2.12.2";
+    version = "2.13.1";
   in
   {
     "${version}-x86_64-darwin" = fetchurl {
-      url = "${base}/stable/release/${version}/sdk/dartsdk-macos-${x86_64}-release.zip";
-      sha256 = "0h6mpy0kfc842vhg053fyxbjnd8lw1d1shdcsj800048260lxhyd";
+      url =
+        "${base}/stable/release/${version}/sdk/dartsdk-macos-${x86_64}-release.zip";
```
Comment (SuperSandro2000) [inline pkgs/development/interpreters/dart/default.nix:18]:
```suggestion
      url = "${base}/stable/release/${version}/sdk/dartsdk-macos-${x86_64}-release.zip";
```

File: pkgs/development/interpreters/dart/default.nix [hunk 3]
```diff
@@ -2,33 +2,36 @@
...
+      sha256 = "0kb6r2rmp5d0shvgyy37fmykbgww8qaj4f8k79rmqfv5lwa3izya";
     };
     "${version}-x86_64-linux" = fetchurl {
-      url = "${base}/stable/release/${version}/sdk/dartsdk-linux-${x86_64}-release.zip";
-      sha256 = "1gg210gf4yif3bl9k19znkndc4c1cd529xwxpi20ykaw3zfxxz2z";
+      url =
+        "${base}/stable/release/${version}/sdk/dartsdk-linux-${x86_64}-release.zip";
```
Comment (SuperSandro2000) [inline pkgs/development/interpreters/dart/default.nix:23]:
```suggestion
      url = "${base}/stable/release/${version}/sdk/dartsdk-linux-${x86_64}-release.zip";
```

File: pkgs/development/interpreters/dart/default.nix [hunk 4]
```diff
@@ -2,33 +2,36 @@
...
+      sha256 = "0zq8wngyrw01wjc5s6w1vz2jndms09ifiymjjixxby9k41mr6jrq";
     };
     "${version}-i686-linux" = fetchurl {
-      url = "${base}/stable/release/${version}/sdk/dartsdk-linux-${i686}-release.zip";
-      sha256 = "1wngxba71j20gq9vy7n8q0m9rnqs047xm5b03bxk3hhaq6dyzkwn";
+      url =
+        "${base}/stable/release/${version}/sdk/dartsdk-linux-${i686}-release.zip";
```
Comment (SuperSandro2000) [inline pkgs/development/interpreters/dart/default.nix:28]:
```suggestion
      url = "${base}/stable/release/${version}/sdk/dartsdk-linux-${i686}-release.zip";
```

File: pkgs/development/interpreters/dart/default.nix [hunk 5]
```diff
@@ -2,33 +2,36 @@
...
+      sha256 = "0zv4q8xv2i08a6izpyhhnil75qhs40m5mgyvjqjsswqkwqdf7lkj";
     };
     "${version}-aarch64-linux" = fetchurl {
-      url = "${base}/stable/release/${version}/sdk/dartsdk-linux-${aarch64}-release.zip";
-      sha256 = "0rqsmzl5g5kgk54qb03kamjm5n5g5pqfl79np37xdzwqbv0zx22b";
+      url =
+        "${base}/stable/release/${version}/sdk/dartsdk-linux-${aarch64}-release.zip";
```
Comment (SuperSandro2000) [inline pkgs/development/interpreters/dart/default.nix:33]:
```suggestion
      url = "${base}/stable/release/${version}/sdk/dartsdk-linux-${aarch64}-release.zip";
```

File: pkgs/development/interpreters/dart/default.nix [hunk 6]
```diff
@@ -40,11 +43,10 @@ stdenv.mkDerivation {
...
   inherit version;
 
-  nativeBuildInputs = [
-    unzip
-  ];
+  nativeBuildInputs = [ unzip ];
 
-  src = sources."${version}-${stdenv.hostPlatform.system}" or (throw "unsupported version/system: ${version}/${stdenv.hostPlatform.system}");
+  src = sources."${version}-${stdenv.hostPlatform.system}" or (throw
+    "unsupported version/system: ${version}/${stdenv.hostPlatform.system}");
```
Comment (SuperSandro2000) [inline pkgs/development/interpreters/dart/default.nix:49]:
```suggestion
  src = sources."${version}-${stdenv.hostPlatform.system}" or (throw "unsupported version/system: ${version}/${stdenv.hostPlatform.system}");
```

File: pkgs/development/interpreters/dart/default.nix [hunk 7]
```diff
@@ -60,7 +62,8 @@ stdenv.mkDerivation {
   meta = with lib; {
     homepage = "https://www.dartlang.org/";
     maintainers = with maintainers; [ grburst thiagokokada ];
-    description = "Scalable programming language, with robust libraries and runtimes, for building web, server, and mobile apps";
+    description =
+      "Scalable programming language, with robust libraries and runtimes, for building web, server, and mobile apps";
```
Comment (SuperSandro2000) [inline pkgs/development/interpreters/dart/default.nix:66]:
```suggestion
    description = "Scalable programming language, with robust libraries and runtimes, for building web, server, and mobile apps";
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 126979` at f3b913e0 run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built successfully:</summary>
  <ul>
    <li>dart</li>
  </ul>
</details>
<details>
  <summary>1 suggestion:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-phase-hooks.md">missing-phase-hooks</a>

`installPhase` should probably contain `runHook preInstall` and `runHook postInstall`.

Near pkgs/development/interpreters/dart/default.nix:46:3:
```
   |
46 |   installPhase = ''
   |   ^
```
</li>
  </ul>
</details>

**Pull request merged**